src/main/java/util/Utility.java

```diff
@@ -328,5 +324,23 @@ public static String getNameClosestToDesiredName(String desiredName, List<String
         return originalSet.stream().map(String::toLowerCase).collect(Collectors.toSet());
     }
 
+    /**
+     * If a value is present in the optional, applies mapping function to it and return the result,
+     * otherwise executes the ifEmpty function and returns an empty optional
+     * @param optional
+     * @param mapper
+     * @param ifEmpty
+     * @return
+     */
+    public static <T, U> Optional<U> flatMapOptional(Optional<T> optional,
```

Comment:
We may need to give this a more specific name to distinguish it from Optional::flatMap. Maybe something with 'log' or 'safe' somewhere? Not sure there's a succinct name...